### PR TITLE
pricehist: init at 1.4.6

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

pricehist is a command-line tool for fetching and formatting historical price data, with support for multiple data sources and output formats. It is usually used with plaintextaccounting tools such as ledger-cli, hledger and beancount, and with GnuCash.

I was able to provide this pull request thanks to the help of @WolfangAukang with #247241.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by WolfangAukang** [comment]
@ofborg eval

**Comment by WolfangAukang** [comment]
FYI, the commit related to `curlify` should be `python3Packages.curlify: init at 2.2.1`

File: pkgs/tools/misc/pricehist/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, buildPythonApplication
+, fetchFromGitLab
+, requests
+, lxml
+, cssselect
+, curlify
+, poetry-core
```
Comment (WolfangAukang) [inline pkgs/tools/misc/pricehist/default.nix:8]:
```suggestion
, poetry-core
, responses
, pytestCheckHook
, pytest-mock
```

**Review by WolfangAukang — CHANGES_REQUESTED** [review]
Pricehist contains tests, so it would be good to apply them when building this package.

I have tested these changes and can confirm all tests are passing.

File: pkgs/tools/misc/pricehist/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+}:
```
Comment (WolfangAukang) [inline pkgs/tools/misc/pricehist/default.nix:9]:
```suggestion
, pytest-mock
, responses
, pytestCheckHook
}:
```

File: pkgs/tools/misc/pricehist/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    lxml
+    cssselect
+    curlify
+    poetry-core
+  ];
+
+  meta = with lib; {
+    description = "A command-line tool for fetching and formatting historical price data, with support for multiple data sources and output formats";
+    homepage = "https://gitlab.com/chrisberkhout/pricehist";
+    licenses = licenses.mit;
```
Comment (WolfangAukang) [inline pkgs/tools/misc/pricehist/default.nix:34]:
```suggestion
    licenses = licenses.mit;
    mainProgram = "pricehist";
```

**Comment by WolfangAukang** [comment]
I cannot test this on `nixpkgs-review` because I get the following:
```
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/bjorn/.cache/nixpkgs-review/pr-253302-3/nixpkgs nixpkgs-overlays=/tmp/tmplynqvbai -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
Nothing to be built.
```

So I had to fetch the PR:
```
> git fetch upstream pull/253302/head:pricehist
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (10/10), done.
remote: Total 14 (delta 2), reused 8 (delta 2), pack-reused 0
Unpacking objects: 100% (14/14), 208.48 KiB | 1.01 MiB/s, done.
From https://github.com/nixos/nixpkgs
 * [new ref]                   refs/pull/253302/head -> pricehist
> git checkout pricehist
Switched to branch 'pricehist'
> nix build .#pricehist
> ./result/bin/pricehist fetch ecb EUR/AUD -s 2023-08-15 -e 2023-08-22
date,base,quote,amount,source,type
2023-08-15,EUR,AUD,1.6888,ecb,reference
2023-08-16,EUR,AUD,1.6895,ecb,reference
2023-08-17,EUR,AUD,1.6931,ecb,reference
2023-08-18,EUR,AUD,1.6999,ecb,reference
2023-08-21,EUR,AUD,1.6995,ecb,reference
2023-08-22,EUR,AUD,1.6878,ecb,reference
> nix build .#python311Packages.curlify
>
```

**Review by WolfangAukang — APPROVED** [review]

**Comment by chrpinedo** [comment]
I am rebasing new commits to the 3 commits with `git push --force` to maintain only the 3 commits in the PR. Perhaps, could it be a problem? should I add new commits and rebase all the commits once we finished working at the PR?

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 253302` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>pricehist</li>
    <li>pricehist.dist</li>
    <li>python310Packages.curlify</li>
    <li>python310Packages.curlify.dist</li>
    <li>python311Packages.curlify</li>
    <li>python311Packages.curlify.dist</li>
  </ul>
</details>

**Pull request merged**